otaclient/app/downloader.py

```diff
@@ -459,3 +458,24 @@ def download(
             compression_alg=compression_alg,
             use_http_if_proxy_set=use_http_if_proxy_set,
         ).result()
+
+    def download_retry_inf(self, *args, **kwargs) -> Tuple[int, int, int]:
+        retry_count = 0
+        inactive_timeout = cfg.DOWNLOAD_GROUP_INACTIVE_TIMEOUT
+        while not self.shutdowned.is_set():
+            try:
+                return self._executor.submit(
+                    self._download_task, *args, **kwargs
+                ).result()
+            except (ExceedMaxRetryError, ChunkStreamingError):
+                if int(time.time()) > self._last_active_timestamp + inactive_timeout:
+                    raise
+
+                retry_count += 1
+                wait_with_backoff(
+                    retry_count,
+                    _backoff_factor=self.BACKOFF_FACTOR,
+                    _backoff_max=self.BACKOFF_MAX,
+                )
+                continue  # infinitely retry on recoverable downloading error
```

Comment:
@Bodong-Yang It should be add log to see what is going on, I think.

Reply:
Updated in [550e873](https://github.com/tier4/ota-client/pull/215/commits/550e873a6a2cae6d8ec1e5f0dd65189d016f2368).

Now if keep failing, the downloader will emit a warning every 60 seconds:
```
logger.warning(f"download {url=} failed for {retry_count+1} times, still keep retry...")
```